src/mesh/mesh_base.C

```diff
@@ -316,6 +318,35 @@ bool MeshBase::has_node_integer(const std::string & name) const
 
 
 void MeshBase::prepare_for_use (const bool skip_renumber_nodes_and_elements, const bool skip_find_neighbors)
+{
+  libmesh_deprecated();
+
+  // We only respect the users wish if they tell us to skip renumbering. If they tell us not to
+  // skip renumbering but someone previously called allow_renumbering(false), then the latter takes
+  // precedence
+  if (skip_renumber_nodes_and_elements)
+    this->allow_renumbering(false);
+
+  // We always accept the user's value for skip_find_neighbors, in contrast to skip_renumber
+  this->allow_find_neighbors(!skip_find_neighbors);
+
+  this->prepare_for_use();
+}
+
+void MeshBase::prepare_for_use (const bool skip_renumber_nodes_and_elements)
+{
+  libmesh_deprecated();
+
+  // We only respect the users wish if they tell us to skip renumbering. If they tell us not to
+  // skip renumbering but someone previously called allow_renumbering(false), then the latter takes
+  // precedence
+  if (skip_renumber_nodes_and_elements)
+    this->allow_renumbering(false);
+
+  this->prepare_for_use();
+}
```

Comment:
These two implementations are no longer actually backwards compatible.  The old behavior, which we're going to revert to (albeit still deprecate) in a new PR shortly, obeys the skip_renumber_nodes_and_elements and skip_find_neighbors values for *one* prepare_for_use call, it doesn't change mesh settings *permanently*.